### ISSUE TITLE
Possible Bug?

### ISSUE THREAD

**Issue by frobnitzem** [lammps/lammps#388]
clang generates the following warning:

../region_block.cpp:404:41: warning: array index 4 is past the end of the array
      (which contains 4 elements) [-Warray-bounds]
    point_on_line_segment(corners[i][3],corners[i][4],x,p);
                                        ^          ~
../region_block.h:39:3: note: array 'corners' declared here
  double corners[6][4][3];
  ^
1 warning generated.

The code in question appears to call point_on_line_segment successively for 0->1, 1->2, 2->3, and then 3->4.  I am not familiar with the code, but maybe 4 should be replaced with 0?

**Comment by timattox** [comment]
That code was added in commit b3d2fb91bb6c959c4c6e045ac95afc3013b7ca76 back in October 2016.  As best I can tell reading the code, you are correct, this is a bug, and will cause reads outside the data structure.  Anyone care to contact the original author of the code to verify changing the 4 to 0 is the correct fix?

**Comment by akohlmey** [comment]
yup. that commit (and the new/changed code it introduced) has already triggered a few issues and we've found and fixed several bugs. I agree, that changing 4 or 0 looks like the logical fix. I've tried all available inputs using this facility and none of them triggers executing the subroutine that contains the code in question. i'll have to look into code coverage, i.e. whether this code is at all reachable and if yes how to trigger it.
